test/ODataReader.v4UnitTests/Given_a_valid_edmx_with_Capability_Annotations.cs

```diff
@@ -0,0 +1,603 @@
+
+            var odcmModel = GetOdcmModel(_edmxElement);
+            var odcmEntityContainer = GetOdcmEntityContainer(odcmModel);
+
+            var odcmEntitySet = odcmEntityContainer.As<OdcmClass>().Properties.Single(p => p.Name == entitySetElementName);
+            OdcmType odcmEntityType = odcmEntitySet.Projection.Type;
+
+            HasOdcmCapability<OdcmUpdateCapability>(odcmEntitySet, updatable)
+                .Should()
+                .BeTrue("Because an entity set with update annotation should have OdcmUpdateCapability");
+
+            odcmEntityType.Projections
+                .Should()
+                .Contain(odcmEntitySet.Projection, "because entity set's Projection is retrieved from its OdcmType's cache of Projections");
+        }
+
+        [Fact]
+        public void When_EntitySet_has_DeleteRestriction_Then_Its_OdcmProperty_has_OdcmDeleteCapability()
+        {
+            var deletable = Any.Bool();
+            var entitySetElement = _entitySetToEntityTypeMapping.Keys.RandomElement();
+            var entitySetElementName = entitySetElement.GetAttribute("Name");
+            entitySetElement.Add(Any.Csdl.DeleteRestrictionAnnotation(deletable, null));
+
+            var odcmModel = GetOdcmModel(_edmxElement);
+            var odcmEntityContainer = GetOdcmEntityContainer(odcmModel);
+
+            var odcmEntitySet = odcmEntityContainer.As<OdcmClass>().Properties.Single(p => p.Name == entitySetElementName);
+            OdcmType odcmEntityType = odcmEntitySet.Projection.Type;
+
+            HasOdcmCapability<OdcmDeleteCapability>(odcmEntitySet, deletable)
+                .Should()
+                .BeTrue("Because an entity set with delete annotation should have OdcmDeleteCapability");
+
+            odcmEntityType.Projections
+                .Should()
+                .Contain(odcmEntitySet.Projection, "because entity set's Projection is retrieved from its OdcmType's cache of Projections");
+        }
+
+        [Fact]
+        public void When_EntitySet_has_ExpandRestriction_Then_Its_OdcmProperty_has_OdcmExpandCapability()
+        {
+            var expandable = Any.Bool();
+            var entitySetElement = _entitySetToEntityTypeMapping.Keys.RandomElement();
+            var entitySetElementName = entitySetElement.GetAttribute("Name");
+            entitySetElement.Add(Any.Csdl.ExpandRestrictionAnnotation(expandable, null));
+
+            var odcmModel = GetOdcmModel(_edmxElement);
+            var odcmEntityContainer = GetOdcmEntityContainer(odcmModel);
+
+            var odcmEntitySet = odcmEntityContainer.As<OdcmClass>().Properties.Single(p => p.Name == entitySetElementName);
+            OdcmType odcmEntityType = odcmEntitySet.Projection.Type;
+
+            HasOdcmCapability<OdcmExpandCapability>(odcmEntitySet, expandable)
+                            .Should()
+                            .BeTrue("Because an entity set with expand annotation should have OdcmExpandCapability");
```

Comment:
I think all navigation properties should have the expand capability and the annotation should determine true vs. false. Thougths?